src/charactersheet/viewmodels/dm/encounter_sections/monster_section/index.js

```diff
@@ -1,309 +1,111 @@
-            return name.toLowerCase().indexOf(term) > -1;
-        });
-        response(results);
-    };
-
-    self.populateMonster = function(label, value) {
-        var monster = DataRepository.monsters[label];
-
-        self.blankMonster().importValues(monster);
-        self.blankMonster().abilityScores().forEach(function(score, idx, _) {
-            score.name(monster.abilityScores[idx].name);
-            score.value(monster.abilityScores[idx].value);
-            score.shortName = ko.observable(score.name().substring(0,3).toUpperCase());
-        });
-        self.showDisclaimer(true);
-    };
-
-    /* Modal Methods */
-
-    self.modalFinishedOpening = function() {
-        self.showDisclaimer(false);
-        self.firstElementInModalHasFocus(true);
-    };
-
-    self.modalFinishedClosing = async function() {
-        self.openModal(false);
-        self.selectPreviewTab();
-
-        if (self.openEditModal()) {
-            const response = await self.currentEditItem().ps.save();
-            Utility.array.updateElement(self.monsters(), response.object, self.editItemIndex);
-        }
-
-        self.openEditModal(false);
-    };
-
-    self.selectPreviewTab = function() {
-        self.convertedDisplayUrl(Utility.string.createDirectDropboxLink(self.currentEditItem().sourceUrl()));
-        self.previewTabStatus('active');
-        self.editTabStatus('');
-    };
-
-    self.selectEditTab = function() {
-        self.editTabStatus('active');
-        self.previewTabStatus('');
-        self.editFirstModalElementHasFocus(true);
-    };
-
-    self.shouldShowExhibitButton = ko.pureComputed(function() {
-        return self._isConnectedToParty();
+    tagline = ko.pureComputed(() => {
+        const index = Fixtures.encounter.sections.monsters.index;
+        return this.encounter().sections()[index].tagline();
     });
 
-    self.pushModalFinishedClosing = function() {
-        self.selectedMonsterToPush(null);
-        self.openPushModal(false);
-    };
+    sorts() {
+        return {
+            ...super.sorts(),
+            'name asc': { field: 'name', direction: 'asc'},
+            'name desc': { field: 'name', direction: 'desc'},
+            'description asc': { field: 'description', direction: 'asc'},
+            'description desc': { field: 'description', direction: 'desc'},
+            'isExhibited asc': { field: 'isExhibited', direction: 'asc'},
+            'isExhibited desc': { field: 'isExhibited', direction: 'desc'},
+        };
+    }
 
-    self.pushModalToPlayerButtonWasPressed = function(monster) {
-        self.selectedMonsterToPush(monster);
-        self.openPushModal(true);
-    };
+    filteredAndSortedEntities = ko.pureComputed(() =>  (
+        SortService.sortAndFilter(this.entities(), this.sort(), null)
+    ), this);
 
-    /* Private Methods */
+    // Actions
 
-    self._dataHasChanged = async function() {
-        if (!self.encounterId()) {
+    async toggleExhibit(monster) {
```

Comment:
as is, would this method:
- mark this as exhibited
- attempt to save the item
- if the item fails, undo marking it as exhibited
- irregardless of success/failure, execute `markAsExhibited` method?

would it be more accurate to wait until you have the result of `await monster.save();` to run both `monster.isExhibited(!monster.isExhibited());` and `this.markAsExhibited(monster.uuid());`?

Reply:
It would never run `markAsExhibited` on a failure because of the return on line 81.

As for the second part, Knockout works diff remember. You have to set `isExhibited` in order for the model state to be saved in the `save` call, but the moment you set that value, KO recomputes the DOM and highlights the row. Hence why you have to watch and then unset it if failure occurs.